transport/conn.go

```diff
@@ -159,6 +160,14 @@ func (c *connReader) loop() {
 	c.bufw = frame.BufferWriter(&c.buf)
 	for {
 		resp := c.recv()
+		if resp.StreamID == eventStreamID {
+			if c.handleEvent != nil {
+				c.handleEvent(resp)
+			} else {
+				log.Printf("unregistered connection received event: %#+v", resp)
+			}
+			continue
```

Comment:
I don't understand. We also send request to close and open control connection again. It is not just logging it.

Reply:
You're of course right, the question rises that if we have some sort of delay between detecting error and closing connection (because of request being on channel waiting to be processed) will further loops in `(c *connReader) loop()` somehow be affected and incorrect?

Reply:
I don't think it causes problems. I changed implementation to prevent stacking multiple loop requests of the same type, so it shouldn't matter.